components/Makefile

```diff
@@ -40,25 +40,24 @@ $(WS_TOP)/components/$(ENCUMBERED)components.mk:	# Makefile # $(WS_TOP)/componen
 
 -include $(WS_TOP)/components/$(ENCUMBERED)components.mk
 
-# depends.mk is auto-generated by concatenating 'depend.mk' files in each
-# component directory.
-$(WS_TOP)/components/$(ENCUMBERED)depends.mk:     $(WS_TOP)/components/$(ENCUMBERED)components.mk
-	@echo "Generating component dependencies..."
-	@cat /dev/null $(shell $(BASS_O_MATIC) --workspace=$(WS_TOP) --components=paths \
-		| sed -e 's;^$(shell pwd)/;;g' \
-		| sed -f components.ignore \
-		| sed -e 's;$$;/depend.mk;g') >$@ \
-		2>/dev/null
-
--include $(WS_TOP)/components/$(ENCUMBERED)depends.mk
-
-# component_mapping.json is auto-generated by the build tools. It provides a map
+# mapping.json is auto-generated by the build tools. It provides a map
 # between component path and component FMRI.
 $(WS_TOP)/components/mapping.json:
 	@echo "Generating component mapping..."
 	@$(WS_TOOLS)/userland-mapping --workspace=$(WS_TOP)
 mapping.json: $(WS_TOP)/components/mapping.json
 
+# dependencies.json is auto-generate by the build tools. It provides information
+# about component dependencies.
+$(WS_TOP)/components/dependencies.json: mapping.json
```

Comment:
Should we support separate dependencies.json for encumbered/*?

Reply:
I left this for simplicity reasons. What would be the benefit of that @pyhalov?

Reply:
The same as for other oi-userland dependencies - answering questions 'what can break if I touch component A'.

Reply:
Well, @pyhalov, but it doesn't matter if you are using single file or multiple files, does it? :)

Reply:
it shouldn't